jetty-server/src/main/java/org/eclipse/jetty/server/handler/ContextHandler.java

```diff
@@ -420,12 +482,31 @@ else if ( _vhosts == null || _vhosts.length == 0)
      * virtual host name. A context with no virtual host names or a null virtual host name is available to all requests that are not served by a context with a
      * matching virtual host name.
      *
-     * @return Array of virtual hosts that this context responds to. A null host name or empty array means any hostname is acceptable. Host names may be String
-     *         representation of IP addresses. Host names may start with '*.' to wildcard one level of names.
-     */
-    @ManagedAttribute(value="Virtual hosts accepted by the context", readonly=true)
+     * @return Array of virtual hosts that this context responds to. A null host name or null/empty array means any hostname is acceptable. Host names may be
+     *         String representation of IP addresses. Host names may start with '*.' to wildcard one level of names. Hosts and wildcard hosts may be followed
+     *         with '@connectorname', in which case they will match only if the the {@link Connector#getName()} for the request also matches. If an entry is
+     *         just '@connectorname' it will match if that connector was used but to match legacy behavior it will only check other '@connectorname' only
+     *         entries for matches. A warning will be logged and host entries will be removed if both host and connector entries are used.
+     * 
+     */
+    @ManagedAttribute(value = "Virtual hosts accepted by the context", readonly = true)
     public String[] getVirtualHosts()
     {
+        if (_vhosts == null)
+            return null;
+
+        String[] vhosts = new String[_vhosts.length];
```

Comment:
You return `_vhosts` below, I think it should be `vhosts`?

Reply:
There was a second issue in here sb.append("*");  instead of sb.append(".*"); it was adding a second . to the hostname